bootstrap/src/main/java/io/airlift/bootstrap/LifeCycleMethodsMap.java

```diff
@@ -23,12 +23,16 @@
 {
     private final Map<Class<?>, LifeCycleMethods> map = Maps.newHashMap();
 
-    synchronized LifeCycleMethods get(Class<?> clazz)
-    {
+    public LifeCycleMethods get(Class<?> clazz) {
         LifeCycleMethods methods = map.get(clazz);
```

Comment:
This is not a safe operation. It's accessing a non-thread-safe mutable object (the map) without proper synchronization. It can cause a variety of issue ranging from getting an incorrect result to looping forever walking the map internal structures.

In general, the double-checked locking idiom is broken due to Java memory model semantics.